source/falaise/snemo/datamodels/base_topology_measurement.h

```diff
@@ -0,0 +1,63 @@
+/// \file falaise/snemo/datamodels/base_topology_measurement.h
+/* Author(s) :    François Mauger <mauger@lpccaen.in2p3.fr>
+ * Creation date: 2012-03-19
+ * Last modified: 2014-01-27
+ *
+ * Description: The base class of topology measurement
+ */
+
+#ifndef FALAISE_SNEMO_DATAMODEL_BASE_TOPOLOGY_MEASUREMENT_H
+#define FALAISE_SNEMO_DATAMODEL_BASE_TOPOLOGY_MEASUREMENT_H 1
+
+// Standard library:
+#include <string>
+
+// Third party:
+// - Bayeux/datatools:
+#include <bayeux/datatools/i_serializable.h>
+#include <bayeux/datatools/i_tree_dump.h>
+#include <bayeux/datatools/properties.h>
+
+namespace snemo {
+
+namespace datamodel {
+
+/// \brief The base class of reconstructed topology
+class base_topology_measurement : public datatools::i_serializable,
+                                  public datatools::i_tree_dumpable {
+ public:
+  /// Constructor
+  base_topology_measurement();
+
+  /// Destructor
+  virtual ~base_topology_measurement();
+
+  /// Return the const container of auxiliaries
+  const datatools::properties& get_auxiliaries() const;
+
+  /// Return the mutable container of auxiliaries
+  datatools::properties& grab_auxiliaries();
+
+  /// Smart print
+  virtual void tree_dump(std::ostream& out_ = std::clog, const std::string& title_ = "",
+                         const std::string& indent_ = "", bool inherit_ = false) const;
+
+ private:
+  datatools::properties _auxiliaries_;  //!< Auxiliary properties
```

Comment:
What is the purpose of the `datatools::properties` data member? The concrete classes don't use it, and it's not clear what information would be stored in it.

Reply:
As you said it is not clear what information would be stored because this is the purpose of these auxiliaries to store non essential information. This prevents the private part of the class to be polluted by additionnal data member. If someone or some algorithm need to store information within this class without breaking the data structure then it has to use this data container.

Reply:
Concrete class of `base_topology_measurement` already define their own private data relevant to them (energy, angle), so there's no pollution. What are the use cases for additional arbitrary information being stored in, e.g. `energy_measurement`? Why not have `base_topology_measurement` as an abstract class and leave concrete classes to add properties data members if required?

Reply:
Ok I agree. Done.